service/cloudconfig/template.go

```diff
@@ -0,0 +1,33 @@
+package cloudconfig
+
+const (
+	getKeyVaultSecretsTemplate = `#!/bin/bash -e
+
+KEY_VAULT_HOST={{ .VaultName }}.vault.azure.net
+AUTH_TOKEN=$(curl http://localhost:50342/oauth2/token --data "resource=https://vault.azure.net" -H Metadata:true | jq -r .access_token)
+API_VERSION=2016-10-01
+
+mkdir -p /etc/kubernetes/ssl/calico
+mkdir -p /etc/kubernetes/ssl/etcd
+
+{{ range $secret := .Secrets }}
+curl https://$KEY_VAULT_HOST/secrets/{{ $secret.SecretName }}?api-version=$API_VERSION -H "Authorization: Bearer $AUTH_TOKEN" | jq -r .value > {{ $secret.FileName }} {{ end }}
+
+chmod 0600 -R /etc/kubernetes/ssl/
+chown -R etcd:etcd /etc/kubernetes/ssl/etcd/
+`
```

Comment:
We discussed we may want to download secrets in separate command of the azure-operator. Then the unit could look like `ExecStart=azure-operator secrets download` or something.

This would be nice because it's Go code, we have imported azure-sdk, and we can implement reliable backoff mechanizm for cheap, and all the code stays in the same repo.

 We should also track it.

Reply:
I like the idea of using Go code for the retry support. A separate command in the operator makes sense to be. Like you say the `azure-sdk` dep is already there. I'll create an issue.

Reply:
Issue is https://github.com/giantswarm/azure-operator/issues/27